Push/Autopush.swift

```diff
@@ -0,0 +1,141 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Common
+import Shared
+import MozillaAppServices
+
+
+protocol AutopushProtocol {
```

Comment:
The idea is that this protocol is held by `Profile` or maybe a different object (if we want Profile to start holding on to less things) so it's easier to mock it that if we attach the Autopush object itself

But if it's more swifty to remove this and just keep the class we can do that!

Reply:
The way you have it now is definitely the more swifty approach 👍